server/lib/routes/match.js

```diff
@@ -1296,69 +1296,55 @@ router.post('/matches', (req, res) => {
-    const bundle = {};
-    bundle.resourceType = 'Bundle';
-    bundle.id = uuid4();
-    bundle.meta = {};
-    bundle.meta.lastUpdated = new Date().toISOString();
-    bundle.type = 'searchset';
-    bundle.total = 0;
-    bundle.entry = []; 
-  
-    const serverUrl = 'http://clientregistry.org/openmrs';
-    const family = {family: ''};
-    const given = {given: ''};
-    const system = {system: 'phone'};
-    const url = {url: 'http://hl7.org/fhir/StructureDefinition/match-grade'};
-    
-    function createPatientEntry(patient, matchGrade, score) {
-      const singleEntry = {};
-      singleEntry.fullUrl = `${serverUrl}/Patient/${patient.id}`;
-      singleEntry.resource = {};
-      singleEntry.resource.resourceType = 'Patient';
-      singleEntry.resource.id = patient.id;
-      singleEntry.resource.meta = {};
-      singleEntry.resource.meta.lastUpdated = new Date().toISOString();
-      singleEntry.resource.meta.source_id = patient.source_id;
-      singleEntry.resource.meta.source = patient.source;
-      singleEntry.resource.gender = patient.gender;
-      singleEntry.resource.name = [];
-      singleEntry.resource.name.push(family);
-      singleEntry.resource.name.push(given);
+  function getPatientById(bundle, array, score, matchGrade) {
+    array.forEach(function(auto) {
+        fhirWrapper.getResource({resource: 'Patient', id: auto.id, noCaching: true }, 
+        (resourceData, statusCode) => {
+            for (const index in resourceData.link) {
+                if (!resourceData.link[index].url) {
+                    continue;
+                }
+                const urlArr = resourceData.link[index].url.split('fhir');
+                if (urlArr.length === 2) {
+                    resourceData.link[index].url = '/fhir' + urlArr[1];
+                }
+            }
 
-      singleEntry.resource.telecom = [];
-      const value = {value: patient.phone};
-      singleEntry.resource.telecom.push(system);
-      singleEntry.resource.telecom.push(value);
+            console.log('The status code is: '+ statusCode);
+            console.log('The patient returned is: '+ JSON.stringify(resourceData));
 
-      singleEntry.resource.birthDate = patient.birthDate;
+            resourceData.search = {
+              extension: [{
+                url: 'http://hl7.org/fhir/StructureDefinition/match-grade',
+                valueCode: matchGrade
+            }],
+            score: score
+            };
 
-      singleEntry.resource.search = {};
-      singleEntry.resource.search.extension = [];
-      const valueCode = {valueCode: matchGrade};
-      singleEntry.resource.search.extension.push(url);
-      singleEntry.resource.search.extension.push(valueCode);
-      singleEntry.resource.search.mode = 'match';
-      singleEntry.resource.search.score = score;
-      return singleEntry;
-    }
+            console.log('After Adding search object to patient object: '+ JSON.stringify(resourceData));
```

Comment:
so far this shows that we are able to build everything as we want and able to add the search object to the patient record retrieved from the fhir server.

As seen:

```
{
  "resourceType": "Patient",
  "id": "443bf26f-4e13-447f-8c58-36db597b3c39",
  "meta": {
    "versionId": "2",
    "lastUpdated": "2024-06-01T17:38:50.119+00:00",
    "source": "#VotjpLeaHo4T0nzZ",
    "profile": [
      "http://b-techbw.github.io/bw-lab-ig/StructureDefinition/bw-patient"
    ],
    "tag": [
      {
        "system": "http://openclientregistry.org/fhir/matchIssues",
        "code": "potentialMatches",
        "display": "Potential Matches"
      },
      {
        "system": "http://openclientregistry.org/fhir/clientid",
        "code": "postman",
        "display": "Postman Testing"
      }
    ]
  },
  "text": {
    "status": "generated",
    "div": "<div xmlns=\"http://www.w3.org/1999/xhtml\"><div class=\"hapiHeaderText\">Aviva Wakelam <b>MATEI </b></div><table class=\"hapiPropertyTable\"><tbody><tr><td>Identifier</td><td>201-01-1235</td></tr><tr><td>Date of birth</td><td><span>15 October 1977</span></td></tr></tbody></table></div>"
  },
  "extension": [
    {
      "url": "patient_status",
      "valueString": "Décédé"
    },
    {
      "url": "patient_status_date",
      "valueDate": "2024-01-01"
    },
    {
      "url": "patient_date_enrollement",
      "valueDate": "2023-12-01"
    },
    {
      "url": "patient_date_Initiation_ARV",
      "valueDate": "2023-12-01"
    }
  ],
  "identifier": [
    {
      "use": "official",
      "system": "https://openmrs.org/UPI",
      "value": "201-01-1235"
    },
    {
      "use": "official",
      "system": "http://openelis-global.org/pat_nationalId"
    },
    {
      "extension": [
        {
          "url": "http://fhir.openmrs.org/ext/patient/identifier#location",
          "valueReference": {
            "reference": "Location/0a2c0967-2a56-41c9-9ad5-0bd959861b42",
            "type": "Location",
            "display": "openmrs-location"
          }
        }
      ],
      "use": "usual",
      "type": {
        "text": "OpenMRS UUID"
      },
      "system": "http://clientregistry.org/openmrs",
      "value": "aeafdcda-d157-44ca-9ac6-e890e0fa3cde"
    }
  ],
  "name": [
    {
      "use": "official",
      "family": "Matei",
      "given": [
        "Aviva",
        "Wakelam"
      ]
    }
  ],
  "telecom": [
    {
      "system": "phone",
      "use": "mobile"
    },
    {
      "system": "phone",
      "use": "home"
    }
  ],
  "gender": "male",
  "birthDate": "1977-10-15",
  "deceasedBoolean": false,
  "link": [
    {
      "other": {
        "reference": "Patient/ea6a99d8-2a58-4686-a8cb-04655646bfae"
      },
      "type": "refer"
    }
  ],
  "search": {
    "extension": [
      {
        "url": "http://hl7.org/fhir/StructureDefinition/match-grade",
        "valueCode": "potential"
      }
    ],
    "score": 0.2
  }
}
```